protocol/contracts/v2/TreasuryPriceIndexOracle.sol

```diff
@@ -91,28 +103,46 @@ contract TreasuryPriceIndexOracle is ITreasuryPriceIndexOracle, TempleElevatedAc
     }
 
     /**
-     * @notice Set the Treasury Price Index (TPI)
-     * @dev 18 decimal places, 1.05e18 == $1.05
+     * @notice The maximum allowed TPI change on any single `setTreasuryPriceIndex()`, in absolute terms.
+     * @dev Used as a bound to avoid unintended/fat fingering when updating TPI
      */
-    function setTreasuryPriceIndex(uint96 value) external override onlyElevatedAccess {
-        // If the cooldownSecs hasn't yet passed since the last update, then this will still
-        // refer to the `previousTpi` value
-        uint96 _oldTpi = treasuryPriceIndex();
-        uint96 _newTpi = value;
+    function setMaxTreasuryPriceIndexTargetDateDelta(uint32 maxTargetDateDelta) external override onlyElevatedAccess {
+        emit MaxTreasuryPriceIndexTargetDateDeltaSet(maxTargetDateDelta);
+        maxTreasuryPriceIndexTargetDateDelta = maxTargetDateDelta;
+    }
 
+    /**
+     * @notice Set the target TPI which will incrementally increase over the given targetDate.
+     * @dev targetDate is unixtime, targetRate is target TPi, 18 decimal places, 1.05e18 == $1.05
+     */
+    function setTreasuryPriceIndexAt(uint96 targetRate, uint32 targetDate) external override onlyElevatedAccess {
+        uint96 _currentTpi = treasuryPriceIndex();
+        uint96 _newTpi = targetRate;
+
+        // Check delta is within margins
         unchecked {
-            uint256 _delta = (_newTpi > _oldTpi) ? _newTpi - _oldTpi : _oldTpi - _newTpi;
-            if (_delta > maxTreasuryPriceIndexDelta) revert BreachedMaxTpiDelta(_oldTpi, _newTpi, maxTreasuryPriceIndexDelta);
+            uint256 _delta = (_newTpi > _currentTpi) ? _newTpi - _currentTpi : _currentTpi - _newTpi;
+            if (_delta > maxTreasuryPriceIndexDelta) revert BreachedMaxTpiDelta(_currentTpi, _newTpi, maxTreasuryPriceIndexDelta);
+
+            if (targetDate + uint32(block.timestamp) < maxTreasuryPriceIndexTargetDateDelta) revert BreachedMaxDateDelta(targetDate, uint32(block.timestamp), maxTreasuryPriceIndexTargetDateDelta);
```

Comment:
Comment from direct commit

> Is this meant to be a >?

Reply:
My understanding is that we wanted to set a minimum threshold (rather than a maximum) so `now + targetDate` must be > `maxTreasuryPriceIndexTargetDateDelta`

Reply:
Should it be called `minTreasuryPriceIndexTargetDateDelta` instead then?

Reply:
Addressed in https://github.com/TempleDAO/temple/pull/1096/commits/d1b959ae1dcd86f6b101d3b5f6f1d853880e33b5

Reply:
This still seems wrong imo.

Lets say:
* targetDate = 1735732800 // 1 Jan 2025
* block.timestamp = 1729857600 // 25 Oct 2024

This is checking `1735732800 + 1729857600 < maxTreasuryPriceIndexTargetDateDelta`

Shouldn't it be:
`if (targetDate - uint32(block.timestamp) < maxTreasuryPriceIndexTargetDateDelta)`

??

Reply:
Fixed and tested in https://github.com/TempleDAO/temple/pull/1096/commits/148c89330e66ed9987bf9c5b0d9f691e329d8cf5